L1Trigger/L1TMuonBarrel/src/L1MuBMTQualPatternLut.cc

```diff
@@ -0,0 +1,214 @@
+//
+//   Class: L1MuDTQualPatternLut
+//
+//   Description: Look-up tables for eta matching unit (EMU)
+//                stores lists of qualified patterns and
+//                coarse eta values 
+//
+//
+//   $Date: 2007/03/30 07:48:02 $
+//   $Revision: 1.1 $
+//
+//   Author :
+//   N. Neumeister            CERN EP
+//   J. Troconiz              UAM Madrid
+//
+//--------------------------------------------------
+
+//-----------------------
+// This Class's Header --
+//-----------------------
+
+#include "L1Trigger/L1TMuonBarrel/interface/L1MuBMTQualPatternLut.h"
+
+//---------------
+// C++ Headers --
+//---------------
+
+#include <iostream>
+#include <iomanip>
+#include <string>
+
+//-------------------------------
+// Collaborating Class Headers --
+//-------------------------------
+
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+#include "CondFormats/L1TObjects/interface/L1TriggerLutFile.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+using namespace std;
+
+// --------------------------------
+//       class L1MuDTQualPatternLut
+//---------------------------------
+
+//----------------
+// Constructors --
+//----------------
+
+L1MuBMTQualPatternLut::L1MuBMTQualPatternLut() {
+
+  //  if ( load() != 0 ) {
+  //    cout << "Can not open files to load eta matching look-up tables for DTTrackFinder!" << endl;
+  //  }
+
+  //  if ( L1MuDTTFConfig::Debug(6) ) print();
+
+}
+
+
+//--------------
+// Destructor --
+//--------------
+
+L1MuBMTQualPatternLut::~L1MuBMTQualPatternLut() {
+
+}
+
+
+//--------------
+// Operations --
+//--------------
+
+//
+// reset look-up tables
+//
+void L1MuBMTQualPatternLut::reset() {
+
+  m_lut.clear();
+
+}
+
+
+//
+// load look-up tables for EMU
+//
+int L1MuBMTQualPatternLut::load() {
+
+  // get directory name
+  string defaultPath ="L1Trigger/"; //"L1TriggerConfig/DTTrackFinder/parameters/";
+    string eau_dir = "L1TMuon/data/bmtf_luts/LUTs_Ass/";//"L1TriggerData/DTTrackFinder/Eau/";
+  string emu_str = "";
+
+  // loop over all sector processors
+  for ( int sp = 0; sp < 6; sp++ ) { 
+    switch ( sp ) {
+      case 0  : { emu_str = "QualPatternList_SP1"; break; }
+      case 1  : { emu_str = "QualPatternList_SP2"; break; }
+      case 2  : { emu_str = "QualPatternList_SP3"; break; }
+      case 3  : { emu_str = "QualPatternList_SP4"; break; }
+      case 4  : { emu_str = "QualPatternList_SP5"; break; }
+      case 5  : { emu_str = "QualPatternList_SP6"; break; }
+    }  
+  
+    // assemble file name
+    edm::FileInPath lut_f = edm::FileInPath(string(defaultPath + eau_dir + emu_str + ".lut"));
+    string emu_file = lut_f.fullPath();
+
+    // open file
+    L1TriggerLutFile file(emu_file);
+    if ( file.open() != 0 ) return -1;
+    //    if ( L1MuDTTFConfig::Debug(1) ) cout << "Reading file : " 
+    //                                         << file.getName() << endl; 
+
+    // ignore comment lines 
+    file.ignoreLines(14);
```

Comment:
this is not terribly robust - you can not follow a usual convention such as ignoring lines that start with a # ?